lib/filesystem/include/filesystem/path.hpp

```diff
@@ -383,6 +384,8 @@ namespace irods::experimental::filesystem
     inline auto swap(path& _lhs, path& _rhs) noexcept -> void { _lhs.swap(_rhs); }
 
     auto hash_value(const path& _p) noexcept -> std::size_t;
+
+    auto get_zone_name(const path& _p) -> std::optional<std::string>;
```

Comment:
maybe this gets renamed to just `zone_name` to fit in with the other functions around it.